tests/helics/system_tests/networkTests.cpp

```diff
@@ -168,4 +168,90 @@ TEST_F(network_tests, test_otherport_fail)
     }
 }
+
+TEST_F(network_tests, test_otherport_env)
+{
+    setEnvironmentalVariable("HELICS_CONNECTION_PORT", "33102");
+    const std::string brokerArgs = "-f 2";
+    auto broker = helics::BrokerFactory::create(helics::core_type::ZMQ, brokerArgs);
+
+    EXPECT_TRUE(broker->isConnected());
+    helics::FederateInfo fi("--core_type=ZMQ --corename=c1");
+    helics::ValueFederate fed1("fed1", fi);
+
+    helics::FederateInfo fi2("--core_type=ZMQ --broker=tcp://127.0.0.1:33102 --corename=c2");
+    helics::ValueFederate fed2("fed2", fi2);
+
+    fed2.enterExecutingModeAsync();
+    fed1.enterExecutingMode();
+    fed2.enterExecutingModeComplete();
+    const auto& address = broker->getAddress();
+    EXPECT_NE(address.find("33102"), std::string::npos);
+    fed1.finalize();
+    fed2.finalize();
+
+    EXPECT_TRUE(broker->waitForDisconnect(std::chrono::milliseconds(400)));
+    if (broker->isConnected()) {
+        broker->disconnect();
+    }
+
+    clearEnvironmentalVariable("HELICS_CONNECTION_PORT");
+}
+
+TEST_F(network_tests, test_otherport_broker_local_env)
+{
+    setEnvironmentalVariable("HELICS_LOCAL_PORT", "33104");
+    const std::string brokerArgs = "-f 2";
+    auto broker = helics::BrokerFactory::create(helics::core_type::ZMQ, brokerArgs);
+    clearEnvironmentalVariable("HELICS_LOCAL_PORT");
+    EXPECT_TRUE(broker->isConnected());
+
+    setEnvironmentalVariable("HELICS_BROKER_PORT", "33104");
+    helics::FederateInfo fi("--core_type=ZMQ --corename=c1b");
+    helics::ValueFederate fed1("fed1", fi);
+    clearEnvironmentalVariable("HELICS_BROKER_PORT");
+
+    helics::FederateInfo fi2("--core_type=ZMQ --broker=tcp://127.0.0.1:33104 --corename=c2b");
+    helics::ValueFederate fed2("fed2", fi2);
+
+    fed2.enterExecutingModeAsync();
+    fed1.enterExecutingMode();
+    fed2.enterExecutingModeComplete();
+    const auto& address = broker->getAddress();
+    EXPECT_NE(address.find("33104"), std::string::npos);
+    fed1.finalize();
+    fed2.finalize();
+    EXPECT_TRUE(broker->waitForDisconnect(std::chrono::milliseconds(400)));
+    if (broker->isConnected()) {
+        broker->disconnect();
+    }
+}
+
+#endif
+
+#ifdef ENABLE_TCP_CORE
+TEST_F(network_tests, test_core_type_env)
+{
+    setEnvironmentalVariable("HELICS_CORE_TYPE", "TCP");
+    const std::string brokerArgs = "-f 2";
+    auto broker = helics::BrokerFactory::create(helics::core_type::TCP, brokerArgs);
+    EXPECT_TRUE(broker->isConnected());
+
+    helics::FederateInfo fi("--corename=c1bt");
+    helics::ValueFederate fed1("fed1", fi);
+
+    helics::FederateInfo fi2("--core_type=TCP --corename=c2bt");
+    helics::ValueFederate fed2("fed2", fi2);
+
+    fed2.enterExecutingModeAsync();
+    fed1.enterExecutingMode();
+    fed2.enterExecutingModeComplete();
+
+    fed1.finalize();
+    fed2.finalize();
+    EXPECT_TRUE(broker->waitForDisconnect(std::chrono::milliseconds(400)));
+    if (broker->isConnected()) {
+        broker->disconnect();
+    }
```

Comment:
Does this need `clearEnvironmentVariable("HELICS_CORE_TYPE");` to prevent issues with other tests that might run after it?

Reply:
In nearly all cases we explicitly set the core type so probably wouldn't cause any issues, but you are right we should clear it.